bundles/org.openhab.binding.nikobus/src/main/java/org/openhab/binding/nikobus/internal/protocol/NikobusConnection.java

```diff
@@ -0,0 +1,153 @@
+    private final Logger logger = LoggerFactory.getLogger(NikobusConnection.class);
+    private final SerialPortManager serialPortManager;
+    private final String portName;
+    private final Consumer<Byte> processData;
+    private @Nullable SerialPort serialPort;
+
+    public NikobusConnection(SerialPortManager serialPortManager, String portName, Consumer<Byte> processData) {
+        this.serialPortManager = serialPortManager;
+        this.portName = portName;
+        this.processData = processData;
+    }
+
+    /**
+     * Return true if this manager is connected.
+     *
+     * @return
+     */
+    public boolean isConnected() {
+        return serialPort != null;
+    }
+
+    /**
+     * Connect to the receiver.
+     *
+     **/
+    public void connect() throws IOException {
+        if (isConnected()) {
+            return;
+        }
+
+        SerialPortIdentifier portId = serialPortManager.getIdentifier(portName);
+        if (portId == null) {
+            throw new IOException("Port is not known!");
+        }
+
+        logger.info("Connecting to {}", portName);
+
+        try {
+            SerialPort serialPort = portId.open("org.openhab.binding.nikobus.pc-link", 2000);
+            serialPort.addEventListener(this);
+            serialPort.notifyOnDataAvailable(true);
+            this.serialPort = serialPort;
+            logger.info("Connected to {}", portName);
+        } catch (PortInUseException e) {
+            throw new IOException("Port is in use!", e);
+        } catch (TooManyListenersException e) {
+            throw new IOException("Cannot attach listener to port!", e);
+        }
+    }
+
+    /**
+     * Closes the connection.
+     **/
+    public void close() {
+        SerialPort serialPort = this.serialPort;
+        this.serialPort = null;
+
+        if (serialPort != null) {
+            try {
+                serialPort.removeEventListener();
+                OutputStream outputStream = serialPort.getOutputStream();
+                if (outputStream != null) {
+                    outputStream.close();
+                }
+                InputStream inputStream = serialPort.getInputStream();
+                if (inputStream != null) {
+                    inputStream.close();
+                }
+            } catch (IOException e) {
+                logger.error("Error closing serial port.", e);
+            } finally {
+                serialPort.close();
+                logger.debug("Closed serial port.");
+            }
+        }
+    }
+
+    /**
+     * Returns an output stream for this connection.
+     */
+    public OutputStream getOutputStream() throws IOException {
+        SerialPort serialPort = this.serialPort;
+        if (serialPort == null) {
+            return null;
+        }
+        return serialPort.getOutputStream();
+    }
+
+    @Override
+    public void serialEvent(SerialPortEvent event) {
+        switch (event.getEventType()) {
```

Comment:
If you have only 1 case it's better to just use an if statements

Reply:
[This](https://github.com/openhab/openhab2-addons/blob/master/bundles/org.openhab.binding.serialbutton/src/main/java/org/openhab/binding/serialbutton/internal/handler/SerialButtonHandler.java) was taken for reference ... updated.